ios/Exponent/Versioned/Modules/Api/EXContacts.m

```diff
@@ -54,13 +67,54 @@ @implementation EXContacts
 
     NSMutableDictionary *contact = [NSMutableDictionary dictionary];
     contact[@"id"] = person.identifier;
+    contact[@"contactType"] = person.contactType == CNContactTypePerson ? @"person" : @"organization";
+    contact[@"name"] = [self _assembleDisplayNameForContact:person];
     contact[@"firstName"] = person.givenName;
     contact[@"lastName"] = person.familyName;
+    contact[@"previousLastName"] = person.previousFamilyName;
     contact[@"middleName"] = person.middleName;
-    contact[@"name"] = [self _assembleDisplayNameForContact:person];
+    contact[@"nickname"] = person.nickname;
     contact[@"company"] = person.organizationName;
     contact[@"jobTitle"] = person.jobTitle;
+    contact[@"department"] = person.departmentName;
+    contact[@"imageAvailable"] = [NSNumber numberWithBool:person.imageDataAvailable];
 
+    if ([keysToFetch containsObject:CNContactNamePrefixKey]) {
+      contact[@"namePrefix"] = person.namePrefix;
+    }
+    if ([keysToFetch containsObject:CNContactNameSuffixKey]) {
+      contact[@"nameSuffix"] = person.nameSuffix;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticGivenNameKey]) {
+      contact[@"phoneticFirstName"] = person.phoneticGivenName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticMiddleNameKey]) {
+      contact[@"phoneticMiddleName"] = person.phoneticMiddleName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticFamilyNameKey]) {
+      contact[@"phoneticLastName"] = person.phoneticFamilyName;
+    }
+    if ([keysToFetch containsObject:CNContactNoteKey]) {
+      contact[@"note"] = person.note;
+    }
+    if ([keysToFetch containsObject:CNContactImageDataKey]) {
+      contact[@"image"] = @{ @"uri": [NSString stringWithFormat:@"%@%@",
```

Comment:
@kmagiera - would `let uri = await Contacts.getImageUriForContactAsync(id);` not be possible on both platforms?

Regardless, your `ContactImage` solution sounds reasonable to me on second thought. `Expo.Contacts.Image` could be a good way to namespace that.